blog/2025-06-15-temporal-impl-1.md

```diff
@@ -0,0 +1,483 @@
+> happens now, is happening now.<br></br> Dark Helmet: What happened to
+> then?<br></br> Colonel Sandurz: We passed then.<br></br> Dark Helmet:
+> When?<br></br> Colonel Sandurz: Just now. We're at now now.<br></br>
+> Dark Helmet: Go back to then.<br></br> Colonel Sandurz: When?<br></br>
+> Dark Helmet: Now.<br></br> Colonel Sandurz: Now?<br></br> Dark Helmet:
+> Now.<br></br> Colonel Sandurz: I can't.<br></br> Dark Helmet:
+> Why?<br></br> Colonel Sandurz: We missed it.<br></br> Dark Helmet:
+> When?<br></br> Colonel Sandurz: Just now.<br></br> Dark Helmet: When
+> will then be now?<br></br> -- <cite>Spaceballs, 1987</cite>
+
+`Temporal.Now` is an incredibly strange type, yet nevertheless
+important. It is the object from which the current instant can be
+measured and mapped into any of the Temporal components.
+
+In JavaScript, this type has no [`[[Construct]]`][construct-link] or
+[`[[Call]]`][call-link] internal method, which is a fancy way to say
+that Now has no constructor and cannot be called directly.
+
+Instead, Now is used primarily as a namespace for its methods.
+
+And this was reflected in early adaptions of Now, which looked more or
+less like the below:
+
+```rust
+struct Now;
+
+impl Now {
+    pub fn instant() -> Instant;
+
+    pub fn zoned_date_time_iso() -> ZonedDateTime;
+}
+```
+
+Interestingly enough, the above implementation is incorrect, or at the
+very least not ideal.
+
+Hidden in the specification steps for Now are some very tricky steps
+invoking the abstract operations: `SystemTimeZoneIdentifier` and
+`SystemUtcEpochNanoseconds`. That's great, let's just use the usual
+suspects `SystemTime` and `iana-time-zone`, merge it, and call it a day
+on the implementation, right?
+
+Except the core purpose of `temporal_rs` is that it can be used in any
+engine implementation, and accessing a system clock and system time zone
+is sometimes difficult for engines that support targets like embedded
+systems. Thus, this functionality must be delegated to the engine or
+runtime ... somehow.
+
+How did we end up implementing `Now` if we have no access to the system
+clock or time zone? Well ... a builder pattern of course!
+
+```rust
+#[derive(Default)]
+pub struct NowBuilder {
+    clock: Option<EpochNanoseconds>,
+    zone: Option<TimeZone>,
+}
+
+impl NowBuilder {
+    pub fn with_system_nanoseconds(mut self, nanoseconds: EpochNanoseconds) -> Self {
+        self.clock = Some(nanoseconds);
+        self
+    }
+
+    pub fn with_system_zone(mut self, zone: TimeZone) -> Self {
+        self.zone = Some(zone);
+        self
+    }
+
+    pub fn build(self) -> Now {
+        Now {
+            clock: self.clock,
+            zone: self.zone.unwrap_or_default(),
+        }
+    }
+}
+
+pub struct Now {
+    clock: Option<EpochNanoseconds>,
+    zone: TimeZone,
+}
+```
+
+Once we've constructed `Now`, then we are off to the races!
+
+In Boa, implementing `Now` is as easy the below implementation for
```

Comment:
Wording seems a bit wonky here. "is as easy the below" specifically

Reply:
Yeah, I tried rewording it in a new commit. Matching it up with  the tone of "off to the races" is a bit odd I think.